File: problems/2025/09-23/queue-reversal/solution.py
from collections import deque

class Solution:
    def reverseQueue(self, q: deque) -> None:
        st = []
        while q:
            st.append(q.popleft())
        while st:
            q.append(st.pop())

if __name__ == "__main__":
    from collections import deque
    q = deque([5,10,15,20,25])
    Solution().reverseQueue(q)
    print(list(q))  # [25,20,15,10,5]

File: problems/2025/09-23/queue-reversal/Solution.java
import java.util.*;
public class Solution {
    public void reverseQueue(Queue<Integer> q) {
        Deque<Integer> st = new ArrayDeque<>();
        while (!q.isEmpty()) st.push(q.remove());
        while (!st.isEmpty()) q.add(st.pop());
    }
    public static void main(String[] args) {
        Queue<Integer> q = new ArrayDeque<>(Arrays.asList(5,10,15,20,25));
        new Solution().reverseQueue(q);
        System.out.println(q); // [25, 20, 15, 10, 5]
    }
}
